dask/array/strategies.py

```diff
@@ -0,0 +1,145 @@
+from __future__ import annotations
+
+try:
+    import hypothesis.strategies as st
+except ImportError as e:
+    raise ImportError(
+        "`dask.array.strategies` requires `hypothesis` to be installed."
+    ) from e
+
+
+@st.composite
+def block_lengths(
+    draw: st.DrawFn,
+    ax_length: int,
+    min_chunk_length: int = 1,
+    max_chunk_length: int | None = None,
+) -> tuple[int, ...]:
+    """Generate different chunking patterns along one dimension of an array."""
+
+    chunks = []
+    remaining_length = ax_length
+    while remaining_length > 0:
+        max_chunk_length_remaining = (
+            min(remaining_length, max_chunk_length)
+            if max_chunk_length
+            else remaining_length
+        )
+
+        if min_chunk_length > max_chunk_length_remaining:
+            # if we are at the end of the array we have no choice but to use a smaller chunk
+            chunk = remaining_length
+        else:
+            chunk = draw(
+                st.integers(
+                    min_value=min_chunk_length, max_value=max_chunk_length_remaining
+                )
+            )
```

Comment:
[Shrinking will tend to work better](https://github.com/HypothesisWorks/hypothesis/blob/master/guides/strategies-that-shrink.rst) if the underlying elements are drawn from the same domain.  In this case I'd implement the strategy by:

1. calculating the range of valid chunk sizes and range of number-of-chunks (excluding the final chunk)
2. drawing from `st.lists(st.integers(...), ...)`
3. discard trailing elements which would take you over `ax_length` and manually insert a final chunk of size `ax_length - sum(chunk_sizes)`

You could even implement this as a non-`@st.composite` function which did some validation and precomputation, and then `return st.lists(...).map(_turn_into_valid_chunks)` to better amortize the setup cost.